sdk/eventhub/Azure.Messaging.EventHubs/src/Consumer/CheckpointPosition.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+using Azure.Core;
+
+namespace Azure.Messaging.EventHubs.Consumer
+{
+    /// <summary>
+    ///   The position in a partition's event stream to use when updating a checkpoint, indicating that an event processor should begin reading from the next event.
+    /// </summary>
+    ///
+    public struct CheckpointPosition
```

Comment:
We should consider implementing `IEquatable<CheckpointPosition>`  for this.  Not only would it be helpful in our tests, but the [Framework Design Guidelines](https://learn.microsoft.com/dotnet/standard/design-guidelines/struct) and [Azure SDK Guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-service-models-basic-data-interfaces) both advocate for it.

Reply:
I was working on updating this, but before we decided to use long.MinValue as the default sequence number value, I was using this a workaround for the old UpdateCheckpoint overload to call the new one, since we don't want to have a `CheckpointPosition(long? sequenceNumber, long? offset)` constructor